test/sys/test_socket.rs

```diff
@@ -255,3 +255,62 @@ pub fn test_syscontrol() {
     // requires root privileges
     // connect(fd, &sockaddr).expect("connect failed");
 }
+
+// Test creating and using named system control sockets
+#[cfg(all(target_os = "linux", target_arch = "x86_64", not(target_env = "musl")))]
+#[test]
+pub fn test_scm_timestamping() {
+    use nix::sys::uio::IoVec;
+    use nix::unistd::close;
+    use nix::sys::socket::{socket, bind, connect, sendmsg, recvmsg, setsockopt,
+                           AddressFamily, SockType, SockFlag, SockAddr,
+                           CmsgSpace, MsgFlags, ControlMessage};
+    use nix::sys::socket::sockopt::{Timestamping};
+    use nix::sys::time::TimeVal;
+    use std::net;
+
+    let actual: net::SocketAddr = FromStr::from_str("127.0.0.1:3000").unwrap();
+    let addr = SockAddr::new_inet(InetAddr::from_std(&actual));
+
+    let fd1 = socket(AddressFamily::Inet, SockType::Datagram, SockFlag::empty(), None).expect("socket failed");
+    connect(fd1, &addr).expect("connect failed");
+
+    let fd2 = socket(AddressFamily::Inet, SockType::Datagram, SockFlag::empty(), None).expect("socket failed");
+    bind(fd2, &addr).expect("binding failed");
+
+    // Definitions from linux/net_tstamp.h
+    const SOF_TIMESTAMPING_SOFTWARE: i32 = 1 << 4;
```

Comment:
These symbols should also be in libc.

Reply:
No, I think it's fine.  libc is not just for POSIX stuff; there's plenty of OS-specific things in there.